mock/py/mockbuild/plugins/rpmautospec.py

```diff
@@ -0,0 +1,117 @@
+            raise ConfigError("The 'rpmautospec_opts.cmd_base' is unset")
+
+        plugins.add_hook("pre_srpm_build", self.attempt_process_distgit)
+        self.log.info("rpmautospec: initialized")
+
+    @traceLog()
+    def attempt_process_distgit(
+        self,
+        host_chroot_spec: Union[Path, str],
+        host_chroot_sources: Optional[Union[Path, str]],
+    ) -> None:
+        """Attempt to process a spec file with rpmautospec."""
+        # Set up variables and check prerequisites.
+        if not host_chroot_sources:
+            self.log.debug("Sources not specified, skipping rpmautospec preprocessing.")
+            return
+
+        host_chroot_spec = Path(host_chroot_spec)
+        host_chroot_sources = Path(host_chroot_sources)
+        if not host_chroot_sources.is_dir():
+            self.log.debug(
+                "Sources not a directory, skipping rpmautospec preprocessing."
+            )
+            return
+
+        distgit_git_dir = host_chroot_sources / ".git"
+        if not distgit_git_dir.is_dir():
+            self.log.debug(
+                "Sources is not a git repository, skipping rpmautospec preprocessing."
+            )
+            return
+
+        host_chroot_sources_spec = host_chroot_sources / host_chroot_spec.name
+        if not host_chroot_sources_spec.is_file():
+            self.log.debug(
+                "Sources doesn’t contain spec file, skipping rpmautospec preprocessing."
+            )
+            return
+
+        with host_chroot_spec.open("rb") as spec, host_chroot_sources_spec.open(
+            "rb"
+        ) as sources_spec:
+            if spec.read() != sources_spec.read():
+                self.log.warning(
+                    "Spec file inside and outside sources are different, skipping rpmautospec"
+                    " preprocessing."
+                )
+                return
+
+        if not specfile_uses_rpmautospec(host_chroot_sources_spec):
+            self.log.debug(
+                "Spec file doesn’t use rpmautospec, skipping rpmautospec preprocessing."
+            )
+            return
+
+        # Install the `rpmautospec` command line tool into the build root.
+        if self.opts.get("requires", None):
+            self.buildroot.pkg_manager.install_as_root(*self.opts["requires"], check=True)
+
+        # Get paths inside the chroot by chopping off the leading paths
+        chroot_dir = Path(self.buildroot.make_chroot_path())
+        chroot_spec = Path("/") / host_chroot_spec.relative_to(chroot_dir)
+        chroot_sources = Path("/") / host_chroot_sources.relative_to(chroot_dir)
+        chroot_sources_spec = Path("/") / host_chroot_sources_spec.relative_to(chroot_dir)
+
+        # Call subprocess to perform the specfile rewrite
+        command = list(self.opts["cmd_base"])
+        command += [chroot_sources_spec]  # <input-spec>
+        command += [chroot_spec]  # <output-spec>
+
+        self.buildroot.doChroot(
```

Comment:
Pretty much the entire reason for doing this in a mock plugin is because we _cannot_ do this on the host. We moved to mock so that we can rely on rpmautospec being available in the target buildroot.